pkg/aws/client/client_route53.go

```diff
@@ -216,40 +224,80 @@ func newResourceRecords(recordType string, values []string) []*route53.ResourceR
-				Name: aws.String(name),
-				Type: aws.String(route53.RRTypeA),
-				AliasTarget: &route53.AliasTarget{
-					DNSName:              resourceRecords[0].Value,
-					HostedZoneId:         aws.String(zoneId),
-					EvaluateTargetHealth: aws.Bool(true),
-				},
+		loadBalanceHostname := aws.StringValue(resourceRecords[0].Value)
+		// if it is a loadbalancer in a known canoncial hosted zone, create resource sets with alias targets for IPv4 and/or IPv6
+		if zoneId := canonicalHostedZoneId(loadBalanceHostname); zoneId != "" {
+			var rrss []*route53.ResourceRecordSet
+			for _, recordType := range GetAliasRecordTypes(loadBalanceHostname) {
+				rrs := &route53.ResourceRecordSet{
+					Name: aws.String(name),
+					Type: aws.String(recordType),
+					AliasTarget: &route53.AliasTarget{
+						DNSName:              &loadBalanceHostname,
+						HostedZoneId:         aws.String(zoneId),
+						EvaluateTargetHealth: aws.Bool(true),
+					},
+				}
+				rrss = append(rrss, rrs)
 			}
+			return rrss
 		}
 	}
-	return &route53.ResourceRecordSet{
-		Name:            aws.String(name),
-		Type:            aws.String(recordType),
-		ResourceRecords: resourceRecords,
-		TTL:             aws.Int64(ttl),
+	return []*route53.ResourceRecordSet{
+		{
+			Name:            aws.String(name),
+			Type:            aws.String(recordType),
+			ResourceRecords: resourceRecords,
+			TTL:             aws.Int64(ttl),
+		},
+	}
+}
+
+func ensureEndsWithDot(host string) string {
+	if strings.HasSuffix(host, ".") {
+		return host
+	}
+	return host + "."
+}
+
+// GetAliasRecordTypes determinate the alias record types needed, depending on if the load balancer has IPv4 and/or IPv6 addresses.
+// The IP addresses of the given hostname are looked by a DNS query.
+func GetAliasRecordTypes(loadbalancerHostname string) []string {
+	ips, err := net.LookupIP(ensureEndsWithDot(loadbalancerHostname))
+	hasIPv4 := err != nil || len(ips) == 0 // assume it has IPv4 addresses if lookup fails temporarily
+	hasIPv6 := err != nil || len(ips) == 0 // assume it has IPv6 addresses if lookup fails temporarily
+	for _, ip := range ips {
+		if ip.To4() != nil {
+			hasIPv4 = true
+		} else {
+			hasIPv6 = true
+		}
+	}
+	var types []string
+	if hasIPv4 {
+		types = append(types, route53.RRTypeA)
 	}
+	if hasIPv6 {
+		types = append(types, route53.RRTypeAaaa)
+	}
+	return types
 }
 
-func getRecordType(recordType, value string) string {
+func isPotentialAliasTarget(recordType, value string) bool {
 	if recordType == route53.RRTypeCname {
 		if zoneId := canonicalHostedZoneId(value); zoneId != "" {
-			return route53.RRTypeA
+			return true
 		}
 	}
-	return recordType
+	return false
 }
 
 var (
 	// original code: https://github.com/kubernetes-sigs/external-dns/blob/master/provider/aws/aws.go
 	// see: https://docs.aws.amazon.com/general/latest/gr/elb.html
+	// Please keep in sync with list in https://github.com/gardener/external-dns-management/blob/master/pkg/controller/provider/aws/aliastarget.go
```

Comment:
Is it worth it to "package" this and consume it, compared to maintaining the copy? The package can be in any of provider-aws/external-dns-management.

Reply:
We would need to refactor external-dns-management first to avoid introducing ghost dependencies.
The list will not change so often and it is not critical to have the latest one.
I will give it a try.

Reply:
After refactoring external-dns-management (https://github.com/gardener/external-dns-management/pull/348), the table is now reused from from there.
@kon-angelo Thanks for the hint!